### PR TITLE
feat: add option to follow system color setting

### LINKED ISSUE #1007: Make theme (light/dark) to follow system preferences
### Clear and concise description of the problem

I want theme to be changed automatically when my Mac OS theme is changing (sunset/sunrise) instead of manually changing theme every day.

### Suggested solution

Add a "Auto" toggle in theme selector, and implement following system's theme preference

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tom-sherman**
Closes #1007

image

### CODE REVIEW

File: components/settings/SettingsColorMode.vue [hunk 1]
```diff
@@ -12,21 +12,29 @@ function setColorMode(mode: ColorMode) {
...
       <div i-ri:sun-line />
       {{ $t('settings.interface.light_mode') }}
     </button>
+    <button
+      btn-text flex-1 flex="~ gap-1 center" p4 border="~ base rounded" bg-base
+      :tabindex="colorMode.preference === 'system' ? 0 : -1"
+      :class="colorMode.preference === 'system' ? 'pointer-events-none' : 'filter-saturate-0'"
+      @click="setColorMode('system')"
+    >
+      {{ $t('settings.interface.system_mode') }}
```
Comment (danielroe) [inline components/settings/SettingsColorMode.vue:37]:
```suggestion
      <div i-ri:computer-line />
      {{ $t('settings.interface.system_mode') }}
```

**Comment by tom-sherman** [comment]
Hmm, this seems bugged when applying the new changes if you already have a saved color preference...

File: locales/en-GB.json [hunk 2]
```diff
@@ -202,7 +202,8 @@
       "default": " (default)",
       "font_size": "Font Size",
       "label": "Interface",
-      "light_mode": "Light Mode"
+      "light_mode": "Light Mode",
+      "system_mode": "Follow system"
```
Comment (danielroe) [inline locales/en-GB.json:206]:
```suggestion
      "system_mode": "System"
```

File: locales/en-US.json [hunk 3]
```diff
@@ -261,6 +261,7 @@
       "font_size": "Font Size",
       "label": "Interface",
       "light_mode": "Light Mode",
+      "system_mode": "Follow system",
```
Comment (danielroe) [inline locales/en-US.json:264]:
```suggestion
      "system_mode": "System",
```

**Comment by tom-sherman** [comment]
Ah, it's not actually a bug with going from main -> this branch but a bug when simply reloading the page...

**Comment by tom-sherman** [comment]
But it works on the preview link!!!! Very strange

**Review by sxzz — APPROVED** [review]

**Pull request merged**